### ISSUE TITLE
Error when checking input: expected lstm_29_input to have shape (None, None, 2) but got array with shape (51, 1, 10)

### ISSUE THREAD

**Issue by tamannasood** [keras-team/keras#12279]
My code is as follows:

tfidf_matrix=dataset[:0,10]

model = Sequential()

input_shape = tfidf_matrix.shape

model.add(LSTM(32, input_dim=2, return_sequences=True,

input_shape=input_shape))

model.add(LSTM(64, input_dim=2, input_shape=input_shape,

return_sequences=True))

model.add(Dropout(0.5))

model.add(Dense(1))

model.add(Activation('sigmoid'))

model.compile(loss='binary_crossentropy', optimizer='rmsprop')

model.fit(X_train, Y_train, batch_size=20, nb_epoch=11)

score = model.evaluate(X_test, Y_test, batch_size=20)

a snippet of my dataset is as follows(It is for a binary classification):

197.2468966,210.0796552,0.579310345,899.0698276,721.1689655,95.52396817,90.59689707,2.620225801,80.53794681,152.6919308,0

200.7320616,213.6853999,0.653851798,856.1059428,841.3118122,97.48691388,92.08916212,2.753934571,100.4151651,161.5956683,1

225.5475209,250.124516,0.735856126,878.6098414,924.2562758,110.4606583,101.748855,2.885445826,77.50833011,109.0694556,0

I'm continuously getting an error somewhat like this: "Error when checking input: expected lstm_9_input to have 3 dimensions, but got array with shape (51, 10)" in model.fit() Kindly help

**Comment by pedro-r-marques** [comment]
I built a very simple example in https://colab.research.google.com:

```
# import tensorflow as tf
from keras.models import Sequential
from keras.layers import LSTM

model = Sequential()
model.add(LSTM(32, input_dim=2, input_shape=(3, 10)))
model.compile(loss='mse', optimizer='sgd')
#model.summary()
model.input
```

This outputs:
```
/usr/local/lib/python2.7/dist-packages/ipykernel_launcher.py:5: UserWarning: The `input_dim` and `input_length` arguments in recurrent layers are deprecated. Use `input_shape` instead.
  """
/usr/local/lib/python2.7/dist-packages/ipykernel_launcher.py:5: UserWarning: Update your `LSTM` call to the Keras 2 API: `LSTM(32, input_shape=(None, 2))`
  """
<tf.Tensor 'lstm_12_input:0' shape=(?, ?, 2) dtype=float32>
```

If I understand correctly, input_dim=2 specifies an input of (unknown batch size, unknown time slots, 2). I would follow the warning above an use input_shape only with (time_slots, vector_size).

Then your inputs to fit (X_train) should have the shape (batch size, time_slots, vector_size).

From the example lines above I'm assuming that time_slots is 10 and vector_size=1.
